src/libraries/System.Private.CoreLib/src/System/Collections/Generic/IReadOnlySet.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Collections.Generic
+{
+    /// <summary>
+    /// Provides a readonly abstraction of a set.
+    /// </summary>
+    /// <typeparam name="T">The type of elements in the set.</typeparam>
+    public interface IReadOnlySet<T> : IReadOnlyCollection<T>
+    {
+        /// <summary>
+        /// Determines if the set contains a specific item
+        /// </summary>
+        /// <param name="item">The item to check if the set contains.</param>
+        /// <returns>true if found; otherwise false.</returns>
+        bool Contains(T item);
+        /// <summary>
+        /// Determines whether the current set is a proper (strict) subset of a specified collection.
+        /// </summary>
+        /// <param name="other">The collection to compare to the current set.</param>
+        /// <returns>true if the current set is a proper subset of other; otherwise false</returns>
+        /// <exception cref="ArgumentNullException">other is null.</exception>
+        bool IsProperSubsetOf(IEnumerable<T> other);
+        /// <summary>
+        /// Determines whether the current set is a proper (strict) superset of a specified collection.
+        /// </summary>
+        /// <param name="other">The collection to compare to the current set.</param>
+        /// <returns>true if the collection is a proper superset of other; otherwise false.</returns>
+        /// <exception cref="ArgumentNullException">other is null.</exception>
+        bool IsProperSupersetOf(IEnumerable<T> other);
+        /// <summary>
+        /// Determine whether the current set is a subset of a specified collection.
+        /// </summary>
+        /// <param name="other">The collection to compare to the current set.</param>
+        /// <returns>true if the current set is a subset of other; otherwise false.</returns>
+        /// <exception cref="ArgumentNullException">other is null.</exception>
+        bool IsSubsetOf(IEnumerable<T> other);
+        /// <summary>
+        /// Determine whether the current set is a super set of a specified collection.
+        /// </summary>
+        /// <param name="other">The collection to compare to the current set</param>
+        /// <returns>true if the current set is a subset of other; otherwise false.</returns>
+        /// <exception cref="ArgumentNullException">other is null.</exception>
+        bool IsSupersetOf(IEnumerable<T> other);
+        /// <summary>
+        /// Determines whether the current set overlaps with the specified collection.
+        /// </summary>
+        /// <param name="other">The collection to compare to the current set.</param>
+        /// <returns>ture if the currnet set and other share at least one common eleemnt; otherwise, false.</returns>
```

Comment:
```suggestion
        /// <returns>true if the current set and other share at least one common element; otherwise, false.</returns>
```

Reply:
@eiriktsarpalis I made a tiny edit to your suggestion.

Reply:
@carlossanlop should I apply these langwords to all instances of true/false while I'm at it?

Reply:
Yes sure, that would be great, thank you. `true`, `false` and also `null`.